python/pyarrow/_compute.pyx

```diff
@@ -473,7 +476,7 @@ cdef class MetaFunction(Function):
 
 cdef _pack_compute_args(object values, vector[CDatum]* out):
     for val in values:
-        if isinstance(val, (list, np.ndarray)):
+        if "numpy" in sys.modules and isinstance(val, (list, np.ndarray)):
```

Comment:
For on the Python/cython side, I think we can have a simple globally used variable like `HAS_NUMPY` that is populated once (eg in lib.pyx), to avoid using the `"numpy" in sys.modules` pattern every time for checking if numpy is available